airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/conftest.py

```diff
@@ -86,9 +86,18 @@ def configured_catalog_fixture(configured_catalog_path, catalog_schemas) -> Opti
 
 
 @pytest.fixture(name="catalog")
-def catalog_fixture(configured_catalog: ConfiguredAirbyteCatalog) -> Optional[AirbyteCatalog]:
+def catalog_fixture(
+    configured_catalog: ConfiguredAirbyteCatalog, docker_runner: ConnectorRunner, connector_config: SecretDict
+) -> Optional[AirbyteCatalog]:
+    """Generates `AirbyteCatalog`, based on `discover` output"""
+
     if configured_catalog:
-        return AirbyteCatalog(streams=[stream.stream for stream in configured_catalog.streams])
+        output = docker_runner.call_discover(config=connector_config)
+        catalog_messages = [message for message in output if message.type == Type.CATALOG]
```

Comment:
After investigating a bit of the code, I got a question, can't we add `primary_key` in the same block where we add `json_schema`? (83 line of this file)

Reply:
That's possible, but I don't think it's the best solution.
`call_discover` output produces catalog of type `AirbyteCatalog`, so we could just filter and return it in `catalog_fixture`.
But `configured_catalog_fixture` produces `ConfiguredAirbyteCatalog`, so if we want to return here the output by the discover, then we need to extract the fields from `AirbyteCatalog` and update `ConfiguredAirbyteCatalog`. The issue there is that `call_discover` output may be changed, updated fields may be changed too, thus we may miss some important data in future.